src/pages/settings/following.vue

```diff
@@ -1,7 +1,154 @@
+              :is-lazy-loaded="true"
+            />
+            <span class="ml-[8px] group-hover:underline">{{
+              follower.displayName | ellipsis
+            }}</span>
+          </NuxtLink>
+          <ButtonV2
+            :preset="
+              unfollowList.includes(follower.wallet) ? 'primary' : 'tertiary'
+            "
+            @click="() => handleClickUnfollow(follower.wallet)"
+          >
+            {{
+              unfollowList.includes(follower.wallet)
+                ? $t('settings_follow_follow')
+                : $t('settings_follow_unfollow')
+            }}
+          </ButtonV2>
+        </div>
+      </div>
+    </CardV2>
+  </div>
+  <div v-else class="flex flex-col justify-center flex-grow">
+    <Label class="text-medium-gray" align="center" :text="$t('settings_following_login_in_label')" />
+    <div class="flex justify-center mt-[24px]">
+      <ProgressIndicator v-if="walletIsLoggingIn" />
+      <ButtonV2
+        v-else
+        :text="$t('settings_following_login_in_button')"
+        preset="secondary"
+        @click="connectWallet"
+      />
+    </div>
+  </div>
+</template>
+
 <script>
+import { mapGetters, mapActions } from 'vuex';
+import walletMixin from '~/mixins/wallet';
+import alertMixin from '~/mixins/alert';
+import { ellipsis } from '~/util/ui';
+
 export default {
-  fetch({ redirect, params, query }) {
-    redirect(302, { name: 'index', params, query });
+  filters: {
+    ellipsis,
+  },
+  mixins: [walletMixin, alertMixin],
+  data() {
+    return {
+      isLoading: false,
+      unfollowList: [],
+    };
+  },
+  computed: {
+    ...mapGetters(['getUserInfoByAddress']),
+    populatedFollowers() {
+      return this.getFollowers.map(follower => ({
+        displayName:
+          this.getUserInfoByAddress(follower)?.displayName || follower,
+        wallet: follower,
+        avatar: this.getUserInfoByAddress(follower)?.avatar,
+      }));
+    },
+  },
+  watch: {
+    walletHasLoggedIn() {
+      if (this.walletHasLoggedIn) {
+        this.fetchFollowerList();
+      }
+    },
+  },
+  async mounted() {
+    if (this.walletHasLoggedIn) {
+      await this.fetchFollowerList();
+    }
+  },
+  methods: {
+    ...mapActions([
+      'lazyGetUserInfoByAddresses',
+      'walletCreatorFollow',
+      'walletCreatorUnfollow',
+    ]),
+    async fetchFollowerList() {
+      if (this.isLoading) {
+        return;
+      }
+      this.isLoading = true;
+      await this.fetchFollowers(this.loginAddress);
+      this.isLoading = false;
+    },
+    async handleClickUnfollow(creator) {
+      try {
+        if (this.unfollowList.includes(creator)) {
+          await this.walletCreatorFollow(creator);
+          const index = this.unfollowList.indexOf(creator);
+          if (index !== -1) {
+            this.unfollowList.splice(index, 1);
+          }
+        } else {
+          await this.walletCreatorUnfollow(creator);
+          this.unfollowList.push(creator);
```

Comment:
What is this `unfollowList` for? I think it is easier to judge by using `populatedFollowers`? In fact does `walletCreatorUnfollow` update `this.getFollowers`? If yes we don't really need a list of `unfollowList` in local scope?

Reply:
The `unfollowList` is for undoing